subset/bacnet/bacnetTests/src/main/java/helper/PicsValidator.java

```diff
@@ -0,0 +1,86 @@
+package helper;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+public class PicsValidator {
+  String read = "R";
+  String write = "W";
+  String optional = "O";
+  String Supported = "TRUE";
+  String formatProperty = "%-35s%-35s%-15s%-25s";
+
+  Multimap<String, String> result = ArrayListMultimap.create();
+
+  boolean testPassed = true;
+
+  public void validate(String bacnetObjectType, String bacnetObjectProperty, String conformanceCode,
```

Comment:
I haven't functionally tested this but:

`testPassed = true` as default.

If anything happens that means your `if`s aren't hit then we may get a false positive. It could be an idea to have the testPassed = false as default - this might be better for debugging especially with a reason for the failure in the device report

Reply:
So the idea is to set testPassed to "true" initially. Once one of the if statement get hit (and has a condition that make the test fail) the "testPassed" boolean is changed to "false" making the overall test to fail (since there is no testPassed = true anywhere).  

I have tested this code with different test cases and has always worked fine. But I get your point and you're definitely right (this is code, anything can happen).

Maybe the thing that is missing is an "else" statement at the very end to make the test fail if none of the if statement are hit.

What do you reckon?

Reply:
Yep, an else would do the same thing and would require less changes to the existing code. Either or, just make sure everything is caught. 

Also:

> this is code _in the wild_ anything can happen